agent/util-scripts/pbench-tool-meister-start

```diff
@@ -514,6 +515,40 @@ class ToolDataSink(BaseServer):
                 break
         return 0 if status == "success" else 1
 
+    @staticmethod
+    def is_running(pid: int) -> bool:
+        """Is the given PID running?
+
+        See https://stackoverflow.com/questions/7653178/wait-until-a-certain-process-knowing-the-pid-end
+
+        Return True if a PID is running, else False if not.
+        """
+        try:
+            os.kill(pid, 0)
+        except ProcessLookupError:
+            return False
+        return True
+
+    @staticmethod
+    def wait_for_pid(pid: int) -> bool:
+        """wait for a process to actually stop running, but eventually give up
+        to avoid a hang."""
+        running = True
+        timeout = time.time() + _TDS_STARTUP_TIMEOUT
+        while running:
+            running = __class__.is_running(pid)
+            if running:
+                if time.time() >= timeout:
+                    break
+                time.sleep(0.1)
+        return running
```

Comment:
This code tests `running`, then overwrites it with a real value, and then tests that; if the process has already stopped running, then we test `running` one more time before we return.

I think we can improve this a bit (in case anyone who is working on this code _now_ is so inclined):
```
        timeout = time.time() + _TDS_STARTUP_TIMEOUT
        while time.time() < timeout:
            if not __class__.is_running(pid):
                return False
            time.sleep(0.1)
        return True
```
In exchange for front-loading the timeout test (and embracing a nested `return` statement), we can get rid of the `running` variable and only test its value once (in the case where the process has already exited).  (It's too bad that Python doesn't offer post-tested loops....)

Reply:
This is not a bad idea, and I intend to consolidate the nearly duplicate code from the start and stop sides anyway. Note, this started as just a direct lift from `pbench-tool-meister-stop`, except I was trying to merge with the old `self.kill()` call which meant either dropping the kill (since we'd never get there without a graceful exit) or adding a timeout. I decided to add a simple timeout, but I didn't worry much about the elegance since this was a patch to plug the current roof leak before I replace the whole roof, and I wanted a really quick turnaround because Riya's PR needed it.